From 93a0204e5b909935b1d5fd4d455dffdcf493d4bd Mon Sep 17 00:00:00 2001
From: darkyzhou <me@zqy.io>
Date: Mon, 20 Jul 2026 22:23:53 +0800
Subject: [PATCH 11/21] AOSCOS: third_party/rust: support stable Rust 1.95

bytemuck 1.25 uses rustversion date checks to select between two portable
SIMD APIs. AOSC OS stable Rust 1.95 already provides the new API, but is
detected as the old API variant. Remove the date check and use the new
implementation.

Signed-off-by: darkyzhou <me@zqy.io>
---
 .../vendor/bytemuck-v1/src/pod.rs                  | 14 --------------
 .../vendor/bytemuck-v1/src/zeroable.rs             | 14 --------------
 2 files changed, 28 deletions(-)

diff --git a/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/pod.rs b/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/pod.rs
index 330f722..3aaee72 100644
--- a/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/pod.rs
+++ b/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/pod.rs
@@ -143,20 +143,6 @@ impl_unsafe_marker_for_simd!(
   }
 );
 
-#[rustversion::before(2026-01-27)] // See https://github.com/Lokathor/bytemuck/issues/343
-#[cfg(feature = "nightly_portable_simd")]
-#[cfg_attr(
-  feature = "nightly_docs",
-  doc(cfg(feature = "nightly_portable_simd"))
-)]
-unsafe impl<T, const N: usize> Pod for core::simd::Simd<T, N>
-where
-  T: core::simd::SimdElement + Pod,
-  core::simd::LaneCount<N>: core::simd::SupportedLaneCount,
-{
-}
-
-#[rustversion::since(2026-01-27)] // See https://github.com/Lokathor/bytemuck/issues/343
 #[cfg(feature = "nightly_portable_simd")]
 #[cfg_attr(
   feature = "nightly_docs",
diff --git a/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/zeroable.rs b/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/zeroable.rs
index 186c567..b0f3b8a 100644
--- a/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/zeroable.rs
+++ b/third_party/rust/chromium_crates_io/vendor/bytemuck-v1/src/zeroable.rs
@@ -222,20 +222,6 @@ impl_unsafe_marker_for_simd!(
     }
 );
 
-#[rustversion::before(2026-01-27)] // See https://github.com/Lokathor/bytemuck/issues/343
-#[cfg(feature = "nightly_portable_simd")]
-#[cfg_attr(
-  feature = "nightly_docs",
-  doc(cfg(feature = "nightly_portable_simd"))
-)]
-unsafe impl<T, const N: usize> Zeroable for core::simd::Simd<T, N>
-where
-  T: core::simd::SimdElement + Zeroable,
-  core::simd::LaneCount<N>: core::simd::SupportedLaneCount,
-{
-}
-
-#[rustversion::since(2026-01-27)] // See https://github.com/Lokathor/bytemuck/issues/343
 #[cfg(feature = "nightly_portable_simd")]
 #[cfg_attr(
   feature = "nightly_docs",
-- 
2.54.0

